### DISCUSSION TITLE
Graphs empty at end of workout

### DISCUSSION THREAD

**Discussion by tylerjames** [cagnulein/qdomyos-zwift#1402] [Q&A]
Just getting started and using QZ to monitor my HR zones and track calories burned, avg power etc. 

Yesterday I did a 1hr Zone 2 workout on an Echelon-EX5. During the whole ride I could see my metrics in QZ and my Apple Watch was connected and sending my heart rate. I was monitoring my zones and power output in real time and I could see the Elapsed Time showing that I had been riding for about an hour. I had QZ running on my Apple Watch as well. 

When I was done I hit Stop in the QZ app. It showed my graphs but they were all empty. Later when I checked Apple Fitness on my phone it had logged the workout and showed the time and avg heart rate but showed nothing for calories. 

I've used it once or twice before and seen the graphs populated so I don't know what I did differently. 

Is there some step that I'm missing?

**Comment by cagnulein** [comment]
hi for sure the issue is that you have to stop first the watch and then qz.
for the chart instead, enable the debug log in the experimental settings,
and if it happens again send me the debug log and i will check it!
thanks!

Il giorno ven 7 apr 2023 alle 18:42 tylerjames ***@***.***>
ha scritto:

> Just getting started and using QZ to monitor my HR zones and track
> calories burned, avg power etc.
>
> Yesterday I did a 1hr Zone 2 workout on an Echelon-EX5. During the whole
> ride I could see my metrics in QZ and my Apple Watch was connected and
> sending my heart rate. I was monitoring my zones and power output on real
> time and I could see the Elapsed Time showing that I had been riding for
> about an hour. I had QZ running on my Apple Watch as well.
>
> When I was done I hit Stop in the QZ app. It showed my graphs but they
> were all empty. Later when I checked Apple Fitness on my phone it had
> logged the workout amd showed the time and avg heart rate but showed
> nothing for calories.
>
> I've used it once or twice before and seen the graphs populated so I don't
> know what I did differently.
>
> Is there some step that I'm missing?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cagnulein/qdomyos-zwift/discussions/1402>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALYWFLTBJSUHDIQDCISY3XAA7V5ANCNFSM6AAAAAAWWXKMSE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud

**Reply by tylerjames** [reply]
Thanks! I thought it might be that. Strangely the Strong app for tracking strength training workouts has a similar problem and it warms you to stop the workout from your watch instead. 

Is there any history kept of past workouts in the app?

**Reply by cagnulein** [reply]
nope but there are the fit files in the folder app that you can upload to strava or enable the auto upload to strava in qz directly! let me know
